### ISSUE TITLE
Sollya tests fail on a simple multiplication core

### ISSUE THREAD

**Issue by monadius** [FPBench/FPBench#91]
Sollya tests fail on the following simple core:
```
(FPCore (arg1 arg2) :precision binary32 (* arg1 arg2))
```
I run Sollya tests with the command:
```
cat sollya.fpcore | racket infra/test-core2sollya.rkt --repeat 2000
```
Usually, there is at least one failed test. For instance:
```
	1626	Fail	(Expected) -8.584786f-39	(Output) -8.584785f-39	(Args) arg1 = 1.06459056f-19
	arg1 = 16473741/154742504910672534362390528, arg2 = -8.063931f-20
	arg2 = -12478329/154742504910672534362390528
```

**Comment by bksaiki** [comment]
Ah, the subnormal bug. This has been on my TODO list for a bit now, but the solution is not so simple. Essentially, the reference interpreter uses Racket's MPFR interface ([math/bigfloat](https://docs.racket-lang.org/math/bigfloat.html)) to compute the result. The most significant flaw in the interface is that it does not properly emulate subnormal arithmetic. The issue surprisingly does not pop up often, but as you've discovered: if you run enough tests, it'll happen regularly.

As I said before, the fix is not trivial. Essentially, we have two options: update the math/bigfloat interface to do subnormal arithmetic, or create a modified interface. As a side effect of some recent work in Herbie, I actually developed an alternate MPFR interface that does subnormal arithmetic, so it shouldn't be too hard to publish a package and require it in the FPBench repo. I currently have a working package locally.

Here is a quick demo using the example you provided:

Racket (bigfloat, `bf-precision = 24`):
```
> (bf* (bf 1.06459056f-19) (bf -8.063931f-20))
(bf "-8.58478529e-39")
```
Generic-flonum (`gfl-exponent = 8`, `gfl-bits = 32`):
```
> (gfl* (gfl 1.06459056f-19) (gfl -8.063931f-20))
#<gfl[8, 32]: -8.58478459e-39>
```
As you can see, my MPFR interface gives the correct result. Hopefully, I can get the package into FPBench soon. Thanks for lodging this issue.

**Comment by pavpanchekha** [comment]
Separating the generic-flonum into its own package is a great idea, @bksaiki. Glad this bug was already on the agenda and thanks to @monadius for filing it.

**Comment by bksaiki** [comment]
Resolved in #94
